torchvision/prototype/datasets/_builtin/gtsrb.py

```diff
@@ -0,0 +1,122 @@
+    }
+    _CHECKSUMS = {
+        "train": "df4144942083645bd60b594de348aa6930126c3e0e5de09e39611630abf8455a",
+        "test": "48ba6fab7e877eb64eaf8de99035b0aaecfbc279bee23e35deca4ac1d0a837fa",
+        "test_gt": "f94e5a7614d75845c74c04ddb26b8796b9e483f43541dd95dd5b726504e16d6d",
+    }
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        rsrcs: List[OnlineResource] = [HttpResource(self._URLS[config.split], sha256=self._CHECKSUMS[config.split])]
+
+        if config.split == "test":
+            rsrcs.append(
+                HttpResource(
+                    self._URLS["test_gt"],
+                    sha256=self._CHECKSUMS["test_gt"],
+                )
+            )
+
+        return rsrcs
+
+    def _filter_images(self, data: Tuple[str, Any]) -> bool:
+        return pathlib.Path(data[0]).suffix == ".ppm"
+
+    def _append_label_train(self, path_and_handle: Tuple[str, Any]) -> Tuple[str, Any, int]:
+        path, handle = path_and_handle
+        label = int(pathlib.Path(path).parent.stem)
+        return path, handle, label
+
+    def _append_label_test(self, path_and_handle: Tuple[str, Any], csv_info: Dict[str, Any]) -> Tuple[str, Any, int]:
+        path, handle = path_and_handle
+        label = int(csv_info["ClassId"])
+        return path, handle, label
+
+    def _collate(
+        self, data: Tuple[str, Any, int], decoder: Optional[Callable[[io.IOBase], torch.Tensor]]
+    ) -> Dict[str, Any]:
+        image_path, image_buffer, label = data
+        return {
+            "image_path": image_path,
+            "image": decoder(image_buffer) if decoder else image_buffer,
+            "label": Label(label),
+        }
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> IterDataPipe[Dict[str, Any]]:
+
+        if config["split"] == "train":
+            images_dp = resource_dps[0]
+            images_dp = Filter(images_dp, self._filter_images)
+            dp = Mapper(images_dp, self._append_label_train)  # path, handle, label
+        else:
+            images_dp, gt_dp = resource_dps
+            dp = Filter(images_dp, self._filter_images)
+
+            gt_dp = CSVDictParser(gt_dp, fieldnames=("Filename", "ClassId"), delimiter=";")
+
+            dp = IterKeyZipper(
+                images_dp,
+                gt_dp,
+                key_fn=path_accessor("name"),
+                ref_key_fn=getitem("Filename"),
+                buffer_size=INFINITE_BUFFER_SIZE,
+                merge_fn=self._append_label_test,
+            )  # path, handle, label
+
+        dp = hint_sharding(dp)
+        dp = hint_shuffling(dp)
+
+        dp = Mapper(dp, partial(self._collate, decoder=decoder))
+        return dp
+
+    def _generate_categories(self, root: pathlib.Path) -> List[str]:
+        config = self.default_config
```

Comment:
The categories are only available in the `"train"` split, correct? Although I don't believe we ever make `split="test"` the default, we can be more specific here

```suggestion
        config = self.info.make_config(split="train")
```